Day4/src/pipeline/extract_employe_data.py

```diff
@@ -0,0 +1,355 @@
+import psycopg2
+
+def connect():
+    return psycopg2.connect(
+        host = "localhost", 
+        database = "extraction_database", 
+        user ="postgres", 
+        password ="password", 
+        port =5432
+        )
+
+def extract_employee_data(cur,con):
+    delete="DELETE FROM raw_employee"
+    cur.execute(delete)
+    con.commit()
+    with open('../sql/extract_employee.sql') as file:
+        sql = " ".join(file.readlines())
+        cur.execute(sql)
+        con.commit()
+
+def extract_timesheet_data(cur,con):
+    delete="DELETE FROM raw_timesheet"
+    cur.execute(delete)
+    con.commit()
+    with open('../sql/extract_timesheet.sql') as file:
+        sql = " ".join(file.readlines())
+        cur.execute(sql)
+        con.commit()
+
+def clean_hours_worked(cur,con):
+    update_hours_worked="UPDATE raw_timesheet SET hours_worked = '0' WHERE paycode ='ABSENT';"
+    cur.execute(update_hours_worked)
+    con.commit()
+
+def create_temporary_punch_details(cur,con):
+    delete="DROP TABLE  IF EXISTS  temporary_punch_details ;"
+    cur.execute(delete)
+    con.commit()
+
+    create_query='''CREATE TABLE temporary_punch_details AS
+        SELECT DISTINCT(employee_id),punch_apply_date as shift_date, MIN(CAST (punch_in_time as TIMESTAMP)) as shift_start_time, 
+        MAX(CAST(punch_out_time as TIMESTAMP)) as shift_end_time
+        FROM raw_timesheet
+        GROUP BY employee_id, punch_apply_date
+        ORDER by 1 asc, 2 asc;'''
+    cur.execute(create_query)
+    con.commit()
+
+def create_temporary_shift_time(cur,con):
+    delete="DROP TABLE  IF EXISTS  temporary_shift_table ;"
+    cur.execute(delete)
+    con.commit()
+    create_query="""
+        CREATE TABLE temporary_shift_table as
+        SELECT employee_id,shift_date,shift_start_time,shift_end_time,
+        CASE 
+        WHEN shift_start_time :: TIMESTAMP :: time BETWEEN '05:00' AND '11:00':: TIME 
```

Comment:
A better approach would be the use of a scalar value function to calculate the shift type. Also, every procedure is in the same file. Try to break it down into smaller parts so that it is easy to maintain.

Reply:
Will keep on note regarding scalar value function from onwards. I will break the procedure.